framework/src/meshgenerators/SubdomainBoundingBoxGenerator.C

```diff
@@ -38,6 +38,9 @@ SubdomainBoundingBoxGenerator::validParams()
       "block_name", "Subdomain name to set for inside/outside the bounding box (optional)");
   params.addParam<MooseEnum>(
       "location", location, "Control of where the subdomain id is to be set");
+  params.addParam<std::vector<SubdomainName>>(
+      "restricted_subdomains",
+      "Only reset subdomain ID for given subdomains within the bounding box");
```

Comment:
Seems like you can set an empty vector as the default and get rid of `_has_restriction`.

Reply:
Then this parameter needs to be called excluding subdomains. Note currently if users set the parameter in the input an empty string, this generator will do nothing.